From 869380d44b2338d59295cf0f6be90cba325b6ba0 Mon Sep 17 00:00:00 2001
Message-Id: <>
In-Reply-To: <>
References: <>
From: Evgeniy Baskov <>
Date: Mon, 13 Mar 2023 15:34:07 +0300
Subject: [PATCH 29/39] efi/libstub: Don't set ramdisk_image/ramdisk_size

The local copy of the boot_params made during build time is used now,
so setting ramdisk_image/ramdisk_size fields is no longer needed,
since they are already set to 0.

Remove no longer required assignments.

Signed-off-by: Evgeniy Baskov <>
---
 drivers/firmware/efi/libstub/x86-stub.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/firmware/efi/libstub/x86-stub.c b/drivers/firmware/efi/libstub/x86-stub.c
index 08d3428fb860..cec866ed9b7a 100644
--- a/drivers/firmware/efi/libstub/x86-stub.c
+++ b/drivers/firmware/efi/libstub/x86-stub.c
@@ -389,9 +389,6 @@ efi_status_t __efiapi efi_pe_entry(efi_handle_t handle,
 	efi_set_u64_split((unsigned long)cmdline_ptr, &hdr->cmd_line_ptr,
 			  &efi_boot_params.ext_cmd_line_ptr);
 
-	hdr->ramdisk_image = 0;
-	hdr->ramdisk_size = 0;
-
 	efi_stub_entry(handle, sys_table_arg, &efi_boot_params);
 	/* not reached */
 
-- 
2.40.1

